### PR TITLE
Add an example for filtering ticket groups

### PR DESCRIPTION
**Pull request by josephspens**
- Adding an example which includes filtering of tickets. There are three filters to modify a list of ticket groups, and when that list updates we update ticket groups in the map.
- Identified a bug in the client library. If `updateTicketGroups` is called before the manifest is fetched, then the update fails silently (none of the sections match). To resolve this, if we have yet to initialize, we are caching the update request and executing it upon initialization (once the manifest is fetched).
- Identified another bug in which sections weren't reset to unavailable prior to repainting the map, leaving sections colored which are no longer included in the available ticket groups
- Identified yet another bug in which resetting a section to unavailable left an opacity on it. Took that off.

### CODE REVIEW

File: examples/filtering-ticket-groups/main.css [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    flex-wrap: wrap;
+}
+
+#filters {
+    width: 100%;
+    display: flex;
+    list-style-type: none;
+    justify-content: center;
+}
+#filters li {
```
Comment (mayavera) [inline examples/filtering-ticket-groups/main.css]: I know it's just an example, but could you add spaces between these rules? It's a little hard on my eyes

File: src/ticketmap/index.tsx [hunk 2]
```diff
@@ -251,15 +259,15 @@ export default class TicketMap extends Component<Props, State> {
     return this.fillSection(sectionId, shouldHighlight)
   }
 
-  selectSection(section: string) {
+  selectSection = (section: string) => {
```
Comment (mayavera) [inline src/ticketmap/index.tsx]: What is this change for?

Comment (josephspens) [reply src/ticketmap/index.tsx]: I was getting errors that functions weren't bound... Some of these functions were called outside the context of the component (via the public API), and I remember you telling me that fat arrow functions don't require explicit binding to the component. I went through and did this for a bunch of functions related to the API just to be safe.

Comment (mayavera) [reply src/ticketmap/index.tsx]: Ah, I see!

File: examples/filtering-ticket-groups/main.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+ */
+
+const fetchTicketGroups = () =>
+  fetch('/examples/data/ticket-groups-1591449.json')
+    .then(response => response.json())
+
+const buildMap = () => {
+  map = new Tevomaps({
+    venueId: '896',
+    configurationId: '14341',
```
Comment (mayavera) [inline examples/filtering-ticket-groups/main.js]: Would you mind defining these values as constants at the top of the file? I tend to fear large files with magic values sprinkled about.

File: Makefile [hunk 4]
```diff
@@ -29,7 +29,7 @@ clean: ## Remove temporary files generated by the build and tests
 clean-all: clean ## Remove all temporary files plus installed dependencies
 	rm -rf ./node_modules
 
-start: install clean ## Start the development server
+start: build ## Start the development server
```
Comment (mayavera) [inline Makefile]: `start` shouldn't rely on `build`. It starts a `webpack-dev-server`, which compiles and serves from memory. It doesn't reference any of the build artifacts.

Comment (mayavera) [reply Makefile]: Also, this seems out of scope

Comment (josephspens) [reply Makefile]: Yes except the examples are using the build files

Comment (mayavera) [reply Makefile]: Ah, I didn't realize `examples/filtering-ticket-groups` was referencing `build/tevomaps.js`. Why not reference `src/index.tsx` instead, as `example/filtering-ticket-groups` is already using webpack?

Comment (mayavera) [reply Makefile]: Also, you can run `eamples/using-lib` without running any commands (building or starting).

**Pull request merged**